db/memtable.cc

```diff
@@ -589,10 +598,14 @@ static bool SaveValue(void* arg, const char* entry) {
       case kTypeSingleDeletion:
       case kTypeRangeDeletion: {
         if (*(s->merge_in_progress)) {
-          *(s->status) = MergeHelper::TimedFullMerge(
-              merge_operator, s->key->user_key(), nullptr,
-              merge_context->GetOperands(), s->value, s->logger, s->statistics,
-              s->env_);
+          *(s->status) = Status::OK();
```

Comment:
if s->pSlice == nullptr then we do not call TimedFulllMerge below and s->status is not updated either. in the below we set found_final_value true and i think it makes sense to update status as well.